hy/compiler.py

```diff
@@ -366,6 +366,17 @@ def is_unpack(kind, x):
             and x[0] == "unpack_" + kind)
 
 
+# Provide asty.Foo(x, ...) as shorthand for
+# ast.Foo(..., lineno=x.start_line, col_offset=x.start_column)
+class Asty(object):
```

Comment:
It has the potential to be used many other times throughout this file. Probably I'll apply it more broadly in a later PR. But I found myself already wanting it, so I saw no reason to wait.